File: paddock/data/fixtures/okf-behavior-go/source/service.go
package dispatch

import "errors"

func Dispatch(items []string, sent *[]string, limit int) (string, int, error) {
	if limit < 0 {
		return "", 0, errors.New("limit must be nonnegative")
	}
	if len(items) == 0 {
		return "empty", 0, nil
	}
	if limit > len(items) {
		limit = len(items)
	}
	selected := items[:limit]
	*sent = append(*sent, selected...)
	return "sent", len(selected), nil
}

File: paddock/data/fixtures/okf-behavior-ts/source/service.ts
export function dispatch(items: string[], sent: string[], limit: number): { status: string; count: number } {
  if (limit < 0) {
    throw new Error("limit must be nonnegative");
  }
  if (items.length === 0) {
    return { status: "empty", count: 0 };
  }
  const selected = items.slice(0, Math.min(limit, items.length));
  sent.push(...selected);
  return { status: "sent", count: selected.length };
}
